common/FileUtils.java

```diff
@@ -243,19 +243,28 @@ public Message readProtoBinaryUnchecked(String path, Message.Builder builder) {
    * Copies a directory to another directory. Creates target directory if needed and doesn't copy
    * symlinks.
    */
-  public void copyDirectoryToDirectory(String source, String destination, String ignored)
+  public void copyDirectoryToDirectory(String source, String destination, String... ignored)
       throws IOException {
     final Path sourcePath = fileSystem.getPath(source);
     final Path targetPath = fileSystem.getPath(destination);
+
     Files.walkFileTree(
         sourcePath,
         new SimpleFileVisitor<Path>() {
           @Override
           public FileVisitResult preVisitDirectory(final Path dir, final BasicFileAttributes attrs)
               throws IOException {
-            if (ignored != null) {
-              if (Pattern.matches(ignored, dir.getFileName().toString())) {
-                return FileVisitResult.CONTINUE;
+            final String dirPath = dir.toAbsolutePath().toString();
```

Comment:
Aha! We are iterating over all files, and we were before as well.
If a folder is ignored, we should return `FileVisitResult.SKIP_SUBTREE`.
I hope this speeds things up.

Reply:
Thanks! It's a really good option.